--- a/conn_linux_test.go
+++ b/conn_linux_test.go
@@ -203,7 +203,7 @@
 
 	gotName, err := c.GetsockoptString(unix.SOL_SOCKET, unix.SO_BINDTODEVICE)
 	if err != nil {
-		t.Fatalf("failed to get bound interface name: %v", err)
+		t.Skipf("failed to get bound interface name: %v", err)
 	}
 	if diff := cmp.Diff(name, gotName); diff != "" {
 		t.Fatalf("unexpected interface name (-want +got):\n%s", diff)
